server/src/main/java/org/candlepin/dto/api/v1/ConsumerDTO.java

```diff
@@ -293,6 +295,58 @@ public ConsumerDTO setUsage(String usage) {
         return this;
     }
 
+    /**
+     * Retrieves the system purpose status for this consumer.
+     *
+     * @return the current system purpose status of this consumer.
+     */
+    public String getSystemPurposeStatus() {
+        return this.systemPurposeStatus;
+    }
+
+    /**
+     * Sets the system purpose status for this ConsumerDTO.
+     *
+     * @param systemPurposeStatus the status value to set.
+     * @return a reference to this DTO object.
+     */
+    public ConsumerDTO setSystemPurposeStatus(String systemPurposeStatus) {
+        this.systemPurposeStatus = systemPurposeStatus;
+        return this;
+    }
+
+    /**
+     * Retrieves the system purpose add ons of this ConsumerDTO object.
+     *
+     * @return the system purpose add ons of the consumer, or null if it has not yet been defined
+     */
+    public Set<String> getAddOns() {
+        return this.addOns != null ? new SetView<>(this.addOns) : null;
+    }
+
+    /**
+     * Sets the system purpose add ons on this ConsumerDTO object.
+     *
+     * @param addOns the add ons to set on this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public ConsumerDTO setAddOns(Set<String> addOns) {
+        if (addOns != null) {
+            if (this.addOns == null) {
+                this.addOns = new HashSet<>();
```

Comment:
Do you think it would make sense to make `addOns` an unmodifiable set via `Collections` since it looks like the set is meant to be cleared on an update instead of modified.

Reply:
I don't think that we can do that since the collection is passed to the SetView and requires the ability to remove elements. As per the contract defined by the DTO framework [here](https://github.com/candlepin/candlepin/blob/master/server/src/main/java/org/candlepin/dto/CandlepinDTO.java#L32), we have to return views.

There's a card to do some work on the views so perhaps we can address your concern then as this is done in other places as well. I'd also like to chat about this with @Ceiu when he's back. Thoughts?